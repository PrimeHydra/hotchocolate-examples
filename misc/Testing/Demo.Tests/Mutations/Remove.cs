namespace Demo.Tests.Mutations;

public class Remove : QueryTestBase
{
    [Fact]
    public async Task RemoveSophie()
    {
        await this.ExecuteQueryAsync(
            @"
mutation {
  removePuppy(input: { id: ""Id_Sophie""}) 
}
                ");

        // Should fail validation
        var result = await this.ExecuteQueryAsync(
            @"
{
  puppy(id: ""Id_Sophie"") {
    name
    breed
  }
}

                   
                ");

        result.MatchSnapshot();
    }
}