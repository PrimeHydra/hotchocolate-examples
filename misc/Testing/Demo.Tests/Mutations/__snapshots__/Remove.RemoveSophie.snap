{
  "errors": [
    {
      "message": "The puppy with ID $Id_Sophie does not exist.",
      "locations": [
        {
          "line": 3,
          "column": 3
        }
      ],
      "path": [
        "puppy"
      ]
    }
  ],
  "data": {
    "puppy": null
  }
}
